src/utils/shift.js

```diff
@@ -51,3 +56,132 @@ export function missingBreaktime({ worktime, breaktime }) {
 export function maxWorktimeExceeded(worktime) {
   return worktime > 10 * 60;
 }
+
+export function maxShifttimeExceeded(shifttime) {
+  return shifttime > 6 * 60;
+}
+
+export class newShiftCutter {
+  new_shift_uuid;
+  shifts;
+
+  constructor(old_shifts, new_shift) {
+    this.new_shift_uuid = new_shift.uuid;
+    this.shifts = old_shifts;
+    if (!Array.isArray(old_shifts)) {
+      this.shifts.push(old_shifts);
+    }
+    if (!this.shifts.find((shift) => shift.uuid === this.new_shift_uuid)) {
+      let new_shift_local = new_shift.clone();
+      new_shift_local.date.start = localizedFormat(
+        new_shift_local.date.start,
+        "yyyy-MM-dd HH:mm:ssXXX",
+        {
+          locale: { localize: "en" }
+        }
+      );
+      new_shift_local.date.end = localizedFormat(
+        new_shift_local.date.end,
+        "yyyy-MM-dd HH:mm:ssXXX",
+        {
+          locale: { localize: "en" }
+        }
+      );
+      new_shift_local.uuid = "000";
+      this.new_shift_uuid = "000";
+      this.shifts.push(new_shift_local);
+    }
+    this.shifts = this.#privateSortShifts();
+  }
+
+  #privateSortShifts() {
+    // todo: delete redundancy when refactor data filter is deployed
+    return this.shifts.sort(
+      (a, b) => parseISO(a.date.start) - parseISO(b.date.start)
+    );
+  }
+
+  calculateCuts() {
```

Comment:
Anmerkung: Dieser Prozess MUSS klipp und klar in Textform mit Aussagekräftigen Beispielen (nicht nur edgecases) dokumentiert werden. Bitte klar in einem Dokument schildern und im Powerfolder ablegen.

Es ist eine Hölle, aber die Implementierung scheint alles abzudecken auch wenn es nicht hübsch ist.
Das zu verbessern steht aber an anderer Stelle an ( #343 )

Reply:
Nicht besser in Code oder auf ReadTheDocs?

Reply:
Solang wir es noch nicht final haben nicht. Das ist ja mit Blick auf #102 wieder fällig